modules/xmpp/SdpTransformUtil.js

```diff
@@ -0,0 +1,439 @@
+        }
+
+        let numSsrcs = getSSRCCount(this.selectedMLine);
+        if (numSsrcs === 1) {
+            // Not using _ssrcs on purpose here
+            return this.selectedMLine.ssrcs[0].id;
+        } else {
+            // Look for a SIM or FID group
+            if (this.selectedMLine.ssrcGroups) {
+                let simGroup = this.findGroup("SIM");
+                if (simGroup) {
+                    return parsePrimarySSRC(simGroup);
+                }
+                let fidGroup = this.findGroup("FID");
+                if (fidGroup) {
+                    return parsePrimarySSRC(fidGroup);
+                }
+            }
+        }
+    }
+
+    /**
+     * Obtains RTX SSRC from the currently selected video media line (the
+     * secondary SSRC of the first "FID" group found)
+     * @param {number} primarySsrc the video ssrc for which to find the
+     * corresponding rtx ssrc
+     * @returns {number|undefined} the rtx ssrc (or undefined if there isn't
+     * one)
+     */
+    getRtxSSRC (primarySsrc) {
+        this._assertMLineSelected();
+
+        let fidGroup = this.findGroupByPrimarySSRC("FID", primarySsrc);
+        return fidGroup ? parseSecondarySSRC(fidGroup) : undefined;
+    }
+
+    /**
+     * Obtains all SSRCs of the currently selected media line.
+     * @return {Array.<number>} an array with all SSRC as numbers.
+     */
+    getSSRCs () {
+        this._assertMLineSelected();
+
+        return this._ssrcs
+            .map(ssrcInfo => ssrcInfo.id)
+            .filter((ssrc, index, array) => array.indexOf(ssrc) === index);
+    }
+
+    /**
+     * Obtains primary video SSRCs (video media line must be selected).
+     * @return {Array.<number>} an array of all primary video SSRCs as numbers.
+     */
+    getPrimaryVideoSSRCs () {
+        this._assertMLineSelected();
+
+        const mediaType = this.selectedMLine.type;
+
+        if (mediaType !== 'video') {
+            throw new Error(
+                "getPrimaryVideoSSRCs doesn't work with '" + mediaType +"'");
+        }
+
+        let videoSSRCs = this.getSSRCs();
+
+        this.forEachSSRCGroup((ssrcGroupInfo) => {
+            // Right now, FID groups are the only ones we parse to
+            // disqualify streams.  If/when others arise we'll
+            // need to add support for them here
+            if (ssrcGroupInfo.semantics === "FID") {
+                // secondary FID streams should be filtered out
+                let secondarySsrc = parseSecondarySSRC(ssrcGroupInfo);
+                videoSSRCs.splice(
+                    videoSSRCs.indexOf(secondarySsrc), 1);
+            }
+        });
+        return videoSSRCs;
+    }
+
+    /**
+     * Checks if there is currently any media selected or throws
+     * an <tt>Error</tt> otherwise.
+     * @private
+     */
+    _assertMLineSelected() {
+        // FIXME if possible wrap all methods instead of adding this call to
```

Comment:
Not sure about the current state, but this should be doable with decorators: https://babeljs.io/docs/plugins/transform-decorators/